chromium_src/chrome/browser/ui/startup/startup_browser_creator.cc

```diff
@@ -25,7 +25,7 @@ class BraveStartupBrowserCreatorImpl final : public StartupBrowserCreatorImpl {
 
   void Launch(Profile* profile,
               chrome::startup::IsProcessStartup process_startup,
-              std::unique_ptr<LaunchModeRecorder> launch_mode_recorder);
+              std::unique_ptr<OldLaunchModeRecorder> launch_mode_recorder);
```

Comment:
how are we going to catch when they add the new launch mode recorder before they removed the old one?

Reply:
We are going to have a build breakage in the call of `StartupBrowserCreatorImpl::Launch`, whenever the new one gets rolled out into `StartupBrowserCreatorImpl`.

Reply:
maybe something like `#ifdef LaunchModeRecorder static_assert(false) #endif`?

Reply:
That might or might not matter. The key here is whatever type is being used as an argument in the signature for `StartupBrowserCreatorImpl::Launch`. I'm gonna add it though as it won't hurt, but it may be the case that at some point we have a new `LaunchModeRecorder` and yet an outdated `StartupBrowserCreatorImpl::Launch` signature.

Reply:
Fixed in c715672b5584f46a9b6325748f5fa52383890722